### PR TITLE
fix typings exports to work with es6 projects

### PR DESCRIPTION
**Pull request by nnance**
@sichvoge this PR will resolve issue #365 and will help to unblock my project once a new release is published to npm.  I added a couple of scripts to the package.json to help contributors work on the project.  The project builds and passes all tests in my environment.

### CODE REVIEW

**Comment by sichvoge** [comment]
Do you know why it fails on Travis?

File: src/raml1/artifacts/raml10parser.ts [hunk 1]
```diff
@@ -1885,6 +1885,14 @@ setAdditionalProperties( param:boolean ){
 
 
         /**
+         * JSON schema style syntax for declaring key restricted maps
+         **/
+patternProperties(  ):TypeDeclaration[]{
```
Comment (sichvoge) [inline src/raml1/artifacts/raml10parser.ts]: @ddenisenko @nnance we need to be careful here. There are no patternProperties in RAML 1.0. See https://github.com/raml-org/raml-js-parser-2/issues/363.

File: src/raml1/artifacts/raml10parserapi.ts [hunk 2]
```diff
@@ -733,6 +733,12 @@ additionalProperties(  ):boolean
 
 
         /**
+         * JSON schema style syntax for declaring key restricted maps
+         **/
+patternProperties(  ):TypeDeclaration[]
```
Comment (sichvoge) [inline src/raml1/artifacts/raml10parserapi.ts]: See my previous comment.

**Comment by sichvoge** [comment]
Thanks @nnance. I added two comments.

**Comment by nnance** [comment]
@sichvoge @ddenisenko sorry but I didn't intend to commit the 2 changes you referenced.  I have removed them now.

**Comment by nnance** [comment]
@sichvoge This is a blocker for my project.  Any idea when this will be merged and published

**Comment by ddenisenko** [comment]
This one will be reviewed today.

**Comment by nnance** [comment]
Thanks

**Comment by ddenisenko** [comment]
Besides whitespace changes, the core of this PR is the following:
1) Remove /typings folder from NPM
2) Remove `/// <reference path="../../typings/main.d.ts" />` from 3 files.

There are 2 issues with this:
1) There are a lot of other references of main.d.ts in other files, so if this approach is chosen, this should be done systematically across the whole code.
2) This may break compilation, because some of those files do actually use typings contributed by main.d.ts. In this particular case compilation probably passes due to main.d.ts being previously visited by the compiler while scanning other files.

All in all this particular approach requires postprocessing of generated `.d.ts` files to remove `/// <reference path="../../typings/main.d.ts" />` from there, not from original `.ts` files.

Tomorrow I'll see if I can either implement this approach and test it with API Workbench in order to make sure it still works with the changes, or if I can find a better way around.

**Comment by nnance** [comment]
@ddenisenko not sure about api workbench but I have built the entire project on my machine just fine and it works with my typescript project as well.  Let me know what you find and I can look into it further if necessary.

**Comment by ddenisenko** [comment]
@nnance I extended and expanded the approach of this PR, removing main.d.ts references automatically from generated .d.ts files, not .ts files. This should make it safer.

Could you please check this PR https://github.com/raml-org/raml-js-parser-2/pull/398 and let me know if parser version built from the code in RemoveTypings branch (with base of hotfix-for-0.2.29 as we cant merge into master at the moment) works for your project?

**Comment by nnance** [comment]
@ddenisenko I can't seem to checkout the remote branch of RemoveTypings or hotfix-for-0.2.29 in order to confirm the fix.  The show up in github but not via the git.  Any ideas?

**Comment by ddenisenko** [comment]
Yes, here is the result of `npm pack`: https://dl.dropboxusercontent.com/u/3776896/raml-1-parser-0.2.29.tgz . It is built against a modified version of typesystem, so dont use it in any sort of production, but should be enough to check if you have things compiling.

Delete raml-1-parser from your node_modules, put the pack into the root and launch `npm install raml-1-parser-0.2.29.tgz`

**Comment by ddenisenko** [comment]
Accidentally merged. Reverted. Sorry, pretty late here.

**Pull request merged**